Mathlib/Order/InitialSeg.lean

```diff
@@ -192,6 +203,48 @@ protected theorem acc (f : r ≼i s) (a : α) : Acc r a ↔ Acc s (f a) :=
     obtain ⟨a', rfl⟩ := f.init hb
     exact ha _ (f.map_rel_iff.mp hb), f.toRelEmbedding.acc a⟩
 
+section PartialOrder
+
+variable [PartialOrder β] {a a' : α} {b : β}
+
+theorem init_le [Preorder α] (f : (· < ·) ≼i (· < ·)) (h : b ≤ f a) : ∃ a', f a' = b := by
```

Comment:
I considered rewriting this as `IsLowerSet (Set.range f)`, but having to explicitly call [`Set.mem_range_self`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Set/Operations.html#Set.mem_range_self) made it a less convenient lemma at the cost of an import.

Reply:
What I'm trying to say is, instead of having the assumption `b ≤ f a`, you'd have something like `∃ x ∈ Set.range f, b ≤ x`, which is certainly less nice.

Reply:
Sure. But then you should rename the lemma to something more discoverable

Reply:
Well, this is matching `init` and `init_iff`. I agree with having a more discoverable name, but I'd rather rename those first.

How about `exists_of_rel` and `exists_iff_rel`? Then this can be `exists_of_le`.

Reply:
`exists_eq_of_rel`, maybe?

Reply:
I perform this rename in #16994, I'll make this a dependent PR.